src/platform/cc13x2_26x2/CC13X2_26X2Config.cpp

```diff
@@ -146,7 +146,13 @@ CHIP_ERROR CC13X2_26X2Config::ReadConfigValue(Key key, uint64_t & val)
 
 CHIP_ERROR CC13X2_26X2Config::ReadConfigValueStr(Key key, char * buf, size_t bufSize, size_t & outLen)
 {
-    return ReadConfigValueBin(key, (uint8_t *) buf, bufSize, outLen);
+    CHIP_ERROR err = ReadConfigValueBin(key, (uint8_t *) buf, bufSize, outLen);
+    if (err == CHIP_NO_ERROR)
+    {
+        buf[outLen] = 0; // since null termination is not stored in NV
```

Comment:
Good catch.

There is a chance `outLen == bufSize` (since [ReadConfigValueBin checks len <= bufSize](https://github.com/ti-tpan/connectedhomeip/blob/03f9947315d68210bd3dde571c3a410a8d9f9ff8/src/platform/cc13x2_26x2/CC13X2_26X2Config.cpp#L165)).

If a caller of `ReadConfigValueStr` does not account for additional null termination, then the null should not be added.

In a broader case, I'm not sure if _every_ caller of `ReadConfigValueStr` needs the null termination. If there is some call which does not require null termination, then we don't need to return the error.
As an example, [GetSerialNumber ](https://github.com/project-chip/connectedhomeip/blob/84c150a1b98746811b4931036d5ac3be0b2c592c/src/include/platform/internal/GenericConfigurationManagerImpl.cpp#L347)assumes null termination, and returns an error accordingly.
^ Does this seem redundant? Should the caller (in this case `GetSerialNumber`) handle `NULL` checks, or should `ReadConfigValueStr`?

So for now, I propose that `ReadConfigValueStr` performs NULL termination with the additional condition that `bufSize > outLen`.

Reply:
Resolved here: https://github.com/project-chip/connectedhomeip/pull/15367/commits/a532260662e3863ce3dc67f5ec68a90a81a1b1af